src/service/url-expander/expander.js

```diff
@@ -245,14 +245,15 @@ export class Expander {
       // If a binding is passed in through opt_bindings it always takes
       // precedence.
       binding = bindingInfo.prioritized;
-    } else if (opt_sync && hasOwn(bindingInfo, 'sync')) {
+    } else if (opt_sync && bindingInfo.sync !== null
```

Comment:
can you explain a bit why `hasOwn` was causing issue?

Reply:
+1. I don't understand either

Reply:
Sure, it confused me too, hence the regression 😕.

It has to do with the way variables are set here (same with `asyncSet`): https://github.com/ampproject/amphtml/blob/a5fc1544598a2344ede79b449fc136604f2eb474/src/service/variable-source.js#L160-L165

An object is created to store both resolvers in this format `{sync: undefined, async: undefined}` . Because of this the `hasOwn` check will be `true` no matter if the if the resolver exists or not. Therefore, it tries to resolve the macro with a non-existent resolver.

Reply:
Ignore this explanation, this is explaining how to fix my mistake, while now I think the original PR should be reverted.